util/test/Test/Pos/Util/Base16.hs

```diff
@@ -0,0 +1,49 @@
+module Test.Pos.Util.Base16
+    ( encode
+    , encodeWithIndex
+    ) where
+
+import           Universum
+
+import qualified Data.ByteString.Base16.Lazy as B16
+import qualified Data.ByteString.Lazy.Char8 as LB
+import           Text.Printf (printf)
+
+-- | Encodes a given ByteString to base-16 and line wraps every 16 bytes.
+encode :: LB.ByteString -> LB.ByteString
+encode = lineWrapBS lineWrapLength . B16.encode
+
+-- | Encodes a given ByteString to base-16 and displays it alongside its byte
+-- offset (line wrapped every 16 bytes).
+encodeWithIndex :: LB.ByteString -> LB.ByteString
+encodeWithIndex bs
+    -- If the length of the ByteString is less than 16 (it hasn't been encoded
+    -- to base-16 yet so we're not checking for <= 32), then just 'encode'
+    -- rather than prepending the byte offsets.
+    | LB.length bs <= (lineWrapLength `div` 2) = encode bs
+    | otherwise = LB.concat $ go 0 (chunkBS lineWrapLength $ B16.encode bs)
+  where
+    go :: Int64 -> [LB.ByteString] -> [LB.ByteString]
+    go _   []     = []
+    go acc (x:xs) =
+        LB.concat
+            [ (LB.pack $ printf "%08x: " acc)
```

Comment:
@parsonsmatt @erikd Opinions on the use of printf?

Reply:
The vast majority of the time, the lengths of bytestrings we're using this for will be up to maybe 1 killobyte, so 8 p;aces on the hex is probably a but much. Would be cool if the sting was less than 256 bytes it only used 2 places and used more places if required.

Reply:
I would be sqeamish about using `print` in production code, especially if it was not by a property test, but in test code I don't have much hesitation.